README.rst

```diff
@@ -13,1837 +13,30 @@ Turn complex GraphQL queries into optimized database queries.
-    that in OrientDB would be of type :code:`A`.
-- If it is impossible to avoid having two abstract OrientDB classes :code:`A` and :code:`B`
-  such that :code:`B` inherits from :code:`A`, you similarly have two options:
-
-  - You may choose to represent :code:`B` as a GraphQL type that can implement the GraphQL interface corresponding
-    to :code:`A`. This makes the GraphQL schema preserve the inheritance relationship between
-    :code:`A` and :code:`B`, but sacrifices the ability for other GraphQL types to inherit from
-    :code:`B`.
-  - You may choose to represent both :code:`A` and :code:`B` as GraphQL interfaces,
-    sacrificing the schema's representation of the inheritance between :code:`A` and :code:`B`, but
-    allowing GraphQL types to inherit from both :code:`A` and :code:`B`. If necessary, you can
-    then create a GraphQL union type :code:`A | B` and use it for any vertex fields that in
-    OrientDB would be of type :code:`A`.
-- It is legal to fully omit classes and fields that are not representable in
-  GraphQL. The compiler currently does not support OrientDB's :code:`EmbeddedMap` type nor
-  embedded non-primitive typed fields, so such fields can simply be omitted in the GraphQL
-  representation of their classes. Alternatively, the entire OrientDB class and all edges that may
-  point to it may be omitted entirely from the GraphQL schema.
-
-Execution model
----------------
-
-Since the GraphQL compiler can target multiple different query
-languages, each with its own behaviors and limitations, the execution
-model must also be defined as a function of the compilation target
-language. While we strive to minimize the differences between
-compilation targets, some differences are unavoidable.
-
-The compiler abides by the following principles:
-
-- When the database is queried with a compiled query string, its response must always be in the
-  form of a list of results.
-- The precise format of each such result is defined by each compilation target separately.
-
-  - :code:`gremlin`, :code:`MATCH` and :code:`SQL` return data in a tabular format, where each
-    result is a row of the table, and fields marked for output are columns.
-  - However, future compilation targets may have a different format. For example,
-    each result may appear in the nested tree format used by the standard
-    GraphQL specification.
-- Each such result must satisfy all directives and types in its corresponding GraphQL query.
-- The returned list of results is **not** guaranteed to be complete!
-
-  - In other words, there may have been additional result sets that satisfy all directives and
-    types in the corresponding GraphQL query, but were not returned by the database.
-  - However, compilation target implementations are encouraged to return complete results if at all
-    practical. The :code:`MATCH` compilation target is guaranteed to produce complete results.
-
-Completeness of returned results
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I moved this section to be a subsection of output_source since the output_source had very little documentation and since this section motivates this directive and explains how to use it.
See https://github.com/kensho-technologies/graphql-compiler/pull/612 for more context.